pandas/core/algorithms.py

```diff
@@ -1305,7 +1305,8 @@ def compute(self, method: str) -> Series:
         narr = len(arr)
         n = min(n, narr)
 
-        kth_val = algos.kth_smallest(arr.copy(), n - 1)
+        # arr passed into kth_smallest must be contiguous
+        kth_val = algos.kth_smallest(arr.copy(order="C"), n - 1)
```

Comment:
could use np.require to only make a copy if needed?

Reply:
The problem is that a copy needs to occur at some point because kth_smallest modifies the input array.. Could instead change kth_smallest to not modify the input array and copy in the algo instead if you think that would be clearer.

Reply:
thanks for explaining, fine as is